x/merkledb/db.go

```diff
@@ -991,7 +991,10 @@ func (db *merkleDB) VerifyChangeProof(
 		return err
 	}
 
-	smallestPath := newPath(start)
+	var smallestPath path
+	if start.hasValue {
```

Comment:
We _could_ just do

```go
smallestPath := newPath(start.value)
```

since the invariant on Maybe says that `value` must be empty but this feels nicer? Idk

Reply:
Ended up deciding to do this